mlflow/store/model_registry/rest_store.py

```diff
@@ -39,29 +41,63 @@
 _logger = logging.getLogger(__name__)
 
 
-class RestStore(AbstractStore):
-    """
-    Note:: Experimental: This entity may change or be removed in a future release without warning.
-    Client for a remote model registry server accessed via REST API calls
+def _get_response_from_method(method):
+    return method.Response()
 
-    :param get_host_creds: Method to be invoked prior to every REST request to get the
-      :py:class:`mlflow.rest_utils.MlflowHostCreds` for the request. Note that this
-      is a function so that we can obtain fresh credentials in the case of expiry.
-    """
 
-    def __init__(self, get_host_creds):
+class BaseRestStore(AbstractStore):  # pylint: disable=abstract-method
```

Comment:
Refactoring out shared logic between the UC and standard `RestStore`s  into an abstract base class

Reply:
did we want to include abstract methods in this ABC to enforce subclass parity (they should override the same methods with the same signatures for both of these, right?) Or is the intention to keep the UC version a bit more flexible?

Reply:
> did we want to include abstract methods in this ABC to enforce subclass parity (they should override the same methods with the same signatures for both of these, right?)

This actually is already enforced transitively from `AbstractRestStore` -> `BaseRestStore` -> subclasses like `UcModelRegistryStore`. If I remove the ` # pylint: disable=abstract-method` from the `UcModelRegistryStore` that inherits from `BaseRestStore`, I get errors like:

```
~/mlflow registered-model-crud !1 ❯ pylint mlflow                                                                                                      Py universe Py 3.8.13 05:00:49 PM
************* Module mlflow.store._unity_catalog.registry.rest_store
mlflow/store/_unity_catalog/registry/rest_store.py:111:0: W0223: Method 'create_model_version' is abstract in class 'AbstractStore' but is not overridden (abstract-method)
```

Reply:
Gotcha! SGTM!